web/modules/custom/joinup_core/src/RequirementsHelper.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\joinup_core;
+
+use Drupal\Core\Database\Connection;
+
+/**
+ * Implements helper methods related to the requirements.
+ */
+class RequirementsHelper {
+
+  /**
+   * The connection class for the primary database storage.
+   *
+   * @var \Drupal\Core\Database\Connection
+   */
+  protected $connection;
+
+  /**
+   * RequirementsHelper constructor.
+   *
+   * @param \Drupal\Core\Database\Connection $connection
+   *   The connection class for the primary database storage.
+   */
+  public function __construct(Connection $connection) {
+    $this->connection = $connection;
+  }
+
+  /**
+   * Returns a query to find the forward revisions.
```

Comment:
This documentation is not correct. This does not return a query, it returns the result of a query. What is meant with "the forward revisions"?

This class is very strange and needs very clear documentation to explain why this code is required that can be understood by future generations which are not familiar with the problem.